src/components/API.js

```diff
@@ -0,0 +1,89 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  /* ------------------ Checks Response from Server ------------------ */
+  _checkServerResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  /* ------------------ Retrieves UserInfo from Server ------------------ */
+  getUserInfo() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    }).then((res) => {
+      return this._checkServerResponse(res);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkServerResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.